contrib/scalar/t/t9099-scalar.sh

```diff
@@ -204,4 +204,35 @@ test_expect_success '`scalar clone` with GVFS-enabled server' '
 	)
 '
 
+test_expect_success '`scalar register` & `unregister` with existing repo' '
+	git init existing &&
```

Comment:
Do you also want to check for a repo that contains a directory? Say, `scalar register existing/folder`? That would help test the parent-peeling logic you added here.

What happens if the repo itself contains a `src` directory (this is common)? What does `scalar register existing/src` do?

And perhaps I'm just not seeing it in the context, do we already test the case of `scalar clone <repo> <path>` followed by `scalar delete <path>` where we don't include the `src` directory?

Reply:
I added a test that covered both (created a repo called `existing` with a `src` subdirectory), which pointed out that the parent-peeling previously didn't happen when `scalar register`/`unregister` was given a path argument. That *should* be fixed now (per the test/manual testing, anyway).

As for the other tests, all of the other ones in this file were for repositories following the `repo_root/src` structure, either via `scalar clone` or manually created. The tests added here are for repositories that are specifically _not_ in a `src` directory.

Reply:
Excellent.